### DISCUSSION TITLE
Make Zipper its own library?

### DISCUSSION THREAD

**Discussion by novaugust** [doorgan/sourceror#90] [Proposals]
Hey doorgan,

Would you have any interest in making Zipper its own library? I have a project that's using it alone from sourceror, and think that if it were a standalone library I might not be the only person who does that :)

(I'd be happy to help maintain it!)

**Comment by doorgan** [comment]
I don't think it would be good to move it to its own library at least for now. Sourceror has zero dev/prod dependencies so it could be easily integrated with ElixirLS, which has issues when runtime dependencies are added to the project(I think it has to do with the pollution they introduce by compiling your project inside the same VM ElixirLS runs on), so while Sourceror is not used in that project today, for the sake of allowing that possibility I want to keep it at zero runtime dependencies. Moving the zipper a standalone library would make sourceror depend on it and complicate that

If it weren't for that I'd be happy to move it to it's own lib

FWIW there is [ZipperEx](https://github.com/hrzndhrn/zipper_ex) as a general purpose zipper inspired by the sourceror implementation, though I believe it doesn't come with an elixir AST zipper implementation

**Comment by novaugust** [comment]
copy that. thanks for the pointer to zipperex
